### PR TITLE
improve error message when using ineligible network with service create

### LINKED ISSUE #24896: [1.12-rc4] Confusing error message when trying to create a service with a bridge network
**Output of `docker version`:**

```
Client:
 Version:      1.12.0-rc4
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   e4a0dbc
 Built:        Thu Jul 14 06:08:44 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-rc4
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   e4a0dbc
 Built:        Wed Jul 13 03:54:54 2016
 OS/Arch:      linux/amd64

```

**Output of `docker info`:**

```
Containers: 13
 Running: 1
 Paused: 0
 Stopped: 12
Images: 59
Server Version: 1.12.0-rc4
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 127
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: active
 NodeID: 8exkt6g1lfazvkcxmier6irmb
 IsManager: Yes
 Managers: 3
 Nodes: 3
 CACertHash: sha256:7b04b668f7faeab34a168cb8739de4af11a591548d40424adda9e433f92df5b3
Runtimes: runc
Default Runtime: runc
Security Options: apparmor
Kernel Version: 3.13.0-92-generic
Operating System: Ubuntu 14.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 488.5 MiB
Name: aws1
ID: 5I6Y:SHUI:K7TT:BWUT:ITQV:65Q3:RYF2:KJK6:3VCG:N3IQ:PSA6:ZAHY
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Labels:
 provider=amazonec2
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Three aws ec2 machines

**Steps to reproduce the issue:**
1. `docker network create mynet`
2. `docker service create --network=mynet alpine`

**Describe the results you received:**

`Error response from daemon: network mynet not found`

**Describe the results you expected:**

A better error message something like `Error response from daemon: network mynet is not swarm-scoped`

### PR DESCRIPTION
**Pull request by mavenugo**
Fixes #24896

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM 🐸
/cc @thaJeztah @tiborvass (put 1.12 milestone but feel free to remove if not needed)

File: daemon/cluster/cluster.go [hunk 1]
```diff
@@ -1075,10 +1075,13 @@ func (c *Cluster) RemoveNetwork(input string) error {
 	return nil
 }
 
-func populateNetworkID(ctx context.Context, c swarmapi.ControlClient, s *types.ServiceSpec) error {
+func (c *Cluster) populateNetworkID(ctx context.Context, client swarmapi.ControlClient, s *types.ServiceSpec) error {
 	for i, n := range s.Networks {
-		apiNetwork, err := getNetwork(ctx, c, n.Target)
+		apiNetwork, err := getNetwork(ctx, client, n.Target)
 		if err != nil {
+			if ln, _ := c.config.Backend.FindNetwork(n.Target); ln != nil && !ln.Info().Dynamic() {
```
Comment (cpuguy83) [inline daemon/cluster/cluster.go]: Seems like a strange place to put this validation, or a strange name for the function.

Comment (mavenugo) [inline daemon/cluster/cluster.go]: I know. I was trying to keep the impact to the minimal by not changing the core functions.
For example, today we have a clear separation between swarm's getNetwork and daemon's FindNetwork. There is no single place today one can get all the networks.

The better fix would be to come up with a elegant function with appropriate filters and context that will differentiate the source of truth & determine eligibility. But I didnt want to invest the effort atm.

Comment (tiborvass) [inline daemon/cluster/cluster.go]: Can `Info()` return nil?

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**